$OpenBSD: patch-cmake_options__linux_cmake,v 1.1.1.1 2022/01/26 19:38:06 kn Exp $

1) remove unknown -fstack-clash-protection (produces warnings spam)

2) We prefer to use the system malloc instead of jemalloc. The problem
that is solved with it has never been reproduced with OpenBSD malloc.
more: https://github.com/telegramdesktop/tdesktop/issues/16084

3) Turn on '-pie' - the bug that caused it to be disabled is solved.
more: https://gitlab.gnome.org/GNOME/nautilus/-/issues/1601

Index: cmake/options_linux.cmake
--- cmake/options_linux.cmake.orig
+++ cmake/options_linux.cmake
@@ -7,7 +7,6 @@
 target_compile_options(common_options
 INTERFACE
     -fstack-protector-all
-    -fstack-clash-protection
     -fPIC
     $<IF:$<CONFIG:Debug>,,-fno-strict-aliasing>
     -pipe
@@ -63,12 +62,6 @@ if (DESKTOP_APP_SPECIAL_TARGET)
     target_link_options(common_options INTERFACE $<IF:$<CONFIG:Debug>,,-g -flto -fuse-linker-plugin>)
 endif()
 
-target_link_libraries(common_options
-INTERFACE
-    desktop-app::external_jemalloc
-    ${CMAKE_DL_LIBS}
-)
-
 if (DESKTOP_APP_USE_ALLOCATION_TRACER)
     target_link_options(common_options
     INTERFACE
@@ -116,6 +109,6 @@ if (NOT DESKTOP_APP_USE_PACKAGED)
         -fwhole-program
         -Wl,-z,relro
         -Wl,-z,now
-        # -pie # https://gitlab.gnome.org/GNOME/nautilus/-/issues/1601
+        -pie
     )
 endif()
